Tests/AWSLambdaEventsTests/Utils/DateWrapperTests.swift

```diff
@@ -18,21 +18,21 @@ import XCTest
 class DateWrapperTests: XCTestCase {
     func testISO8601CodingWrapperSuccess() {
         struct TestEvent: Decodable {
-            @ISO8601Coding
-            var date: Date
+            // @ISO8601Coding
+            var date: ISO8601Coding
         }
 
         let json = #"{"date":"2020-03-26T16:53:05Z"}"#
         var event: TestEvent?
         XCTAssertNoThrow(event = try JSONDecoder().decode(TestEvent.self, from: json.data(using: .utf8)!))
 
-        XCTAssertEqual(event?.date, Date(timeIntervalSince1970: 1_585_241_585))
+        XCTAssertEqual(event?.date.wrappedValue, Date(timeIntervalSince1970: 1_585_241_585))
```

Comment:
I'm not a big fan, of the `date.wrappedValue` approach. IMHO how the date decoding is done should be an implementation detail and nothing that is exposed to the enduser in this way (forced to use `.wrappedValue`). 

If we want to support `5.0` I would love to see the decoding implement in the structs itself. (we can just copy and paste code from my old repo).

@tomerd wdyt? I could do this...

Reply:
hi @fabianfett the end user is not exposed to wrappedValue, this is only exposed in the tests. here is an example of the impl

```
public struct Message {
        public enum Attribute {
            case string(String)
            case binary([UInt8])
        }

        public let signature: String
        public let messageId: String
        public let type: String
        public let topicArn: String
        public let messageAttributes: [String: Attribute]
        public let signatureVersion: String
        public let signingCertURL: String
        private var _timestamp: ISO8601WithFractionalSecondsCoding
        public var timestamp: Date {
            get { return self._timestamp.wrappedValue }
            set { self._timestamp = ISO8601WithFractionalSecondsCoding(wrappedValue: newValue) }
        }

        public let message: String
        public let unsubscribeUrl: String
        public let subject: String?
    }
```